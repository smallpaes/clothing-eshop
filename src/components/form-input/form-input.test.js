import { shallow } from 'enzyme';

import FormInput from './form-input.component';

describe('FormInput component', () => {
  let wrapper;
  let mockHandleChange;
  const mockLabel = 'dummy label';

  beforeEach(() => {
    mockHandleChange = jest.fn();
    const mockProps = {
      handleChange: mockHandleChange,
      label: mockLabel,
      value:'test'
    };

    wrapper = shallow(<FormInput { ...mockProps }/>);
  });

  it('Should render FormInput component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should trigger handleChange once upon FormInputContainer onChange', () => {
    wrapper.find('FormInputContainer').simulate('change');
    expect(mockHandleChange.mock.calls.length).toBe(1);
  });

  it('Should render FormInputLabel when label prop is passed in', () => {
    expect(wrapper.exists('FormInputLabel')).toBe(true);
  });

  it('Should not render FormInputLabel when label prop in not passed in', () => {
    const newWrapper = shallow(<FormInput handleChange={mockHandleChange} />);
    expect(newWrapper.exists('FormInputLabel')).toBe(false);
  });

  it('Should has shrink as a class when value prop is string', () => {
    expect(wrapper.find('FormInputLabel').prop('className')).toBe('shrink');
  });

  it('Should has empty class value when value is number', () => {
    const mockNewProps = {
      handleChange: mockHandleChange, 
      label: mockLabel,
      value: 1
    }
    const newWrapper = shallow(<FormInput { ...mockNewProps } />);
    expect(newWrapper.find('FormInputLabel').prop('className')).toBe('');
  });
})