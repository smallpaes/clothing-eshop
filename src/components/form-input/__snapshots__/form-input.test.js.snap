// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component Should render FormInput component 1`] = `
<styled.div>
  <FormInputContainer
    onChange={[MockFunction]}
    value="test"
  />
  <FormInputLabel
    className="shrink"
  >
    dummy label
  </FormInputLabel>
</styled.div>
`;
